src/main/java/gov/nih/nci/evs/api/service/SparqlQueryCacheService.java

```diff
@@ -63,19 +65,26 @@ public List<String> getHierarchy(
 
     final Sparql sparqlResult = mapper.readValue(res, Sparql.class);
     final Bindings[] bindings = sparqlResult.getResults().getBindings();
+    final Set<String> seen = new HashSet<>();
     for (final Bindings b : bindings) {
       final StringBuffer str = new StringBuffer();
-      str.append(
+      final String par =
           b.getParentCode() == null
               ? EVSUtils.getCodeFromUri(b.getParent().getValue())
-              : b.getParentCode().getValue());
+              : b.getParentCode().getValue();
```

Comment:
We are using duplicated code here as well to fill out the string. We should move away from having duplicated code for easier code management and instead create a help method to call. That way we only need to update in one place

Reply:
@bcarlsenca